### ISSUE TITLE
Code style analyzers package doesn't load global config files on Linux

### ISSUE THREAD

**Issue by ArturDorochowicz** [dotnet/roslyn#58948]
**Version Used**:  [Microsoft.CodeAnalysis.CSharp.CodeStyle 4.0.1](https://www.nuget.org/packages/Microsoft.CodeAnalysis.CSharp.CodeStyle/4.0.1)

The package doesn't load its own predefined globalconfig files on case sensitive file systems.

The package contains the following relevant files:
```
build
   config
      AnalysisLevelStyle_None.editorconfig
      AnalysisLevelStyle_Recommended.editorconfig
      AnalysisLevelStyle_All.editorconfig
      AnalysisLevelStyle_Default.editorconfig 
      AnalysisLevelStyle_Minimum.editorconfig
   Microsoft.CodeAnalysis.CSharp.CodeStyle.targets
   Microsoft.CodeAnalysis.CSharp.CodeStyle.props"
```
The code in `Microsoft.CodeAnalysis.CSharp.CodeStyle.targets` that loads the config file,  builds the path with `ToLowerInvariant()` and therefore doesn't load anything on case sensitive file systems, e.g. on Linux:
```xml
<_GlobalAnalyzerConfigFileName_MicrosoftCodeAnalysisCSharpCodeStyle>AnalysisLevelStyle_$(_GlobalAnalyzerConfigAnalysisMode_MicrosoftCodeAnalysisCSharpCodeStyle).editorconfig</_GlobalAnalyzerConfigFileName_MicrosoftCodeAnalysisCSharpCodeStyle>
<_GlobalAnalyzerConfigFileName_MicrosoftCodeAnalysisCSharpCodeStyle>$(_GlobalAnalyzerConfigFileName_MicrosoftCodeAnalysisCSharpCodeStyle.ToLowerInvariant())</_GlobalAnalyzerConfigFileName_MicrosoftCodeAnalysisCSharpCodeStyle> 
```

**Steps to Reproduce**:

1. `dotnet new globaljson --sdk-version 6.0.101`
2. `dotnet new web`
3. `dotnet add package Microsoft.CodeAnalysis.CSharp.CodeStyle --version 4.0.1`
4. `dotnet build -p:AnalysisLevelStyle=Recommended -p:AnalysisLevelSuffixStyle=Recommended`

On Linux this incorrectly runs with no warnings:
```
dotnet build -p:AnalysisLevelStyle=Recommended -p:AnalysisLevelSuffixStyle=Recommended
Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored /home/ubuntu/bug1/bug1.csproj (in 222 ms).
  bug1 -> /home/ubuntu/bug1/bin/Debug/net6.0/bug1.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:03.42
```

On Windows the build has warnings, as expected:
```
Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.
 
  Determining projects to restore...
  Restored [...]\bug1\bug1.csproj (in 298 ms).
[...]\bug1\Program.cs(1,1): warning IDE0008: Use explicit type instead of 'var' [[...]\bug1\bug1.csproj]
[...]\bug1\Program.cs(2,1): warning IDE0008: Use explicit type instead of 'var' [[...]\bug1\bug1.csproj]
  bug1 -> [...]\bin\Debug\net6.0\bug1.dll

Build succeeded.
[...]\bug1\Program.cs(1,1): warning IDE0008: Use explicit type instead of 'var' [[...]\bug1\bug1.csproj]
[...]\bug1\Program.cs(2,1): warning IDE0008: Use explicit type instead of 'var' [[...]\bug1\bug1.csproj]
    2 Warning(s)
    0 Error(s)

Time Elapsed 00:00:02.41
```

**Expected Behavior**:
The package should correctly load the specified config file under Linux. Just like it does on Windows. 

**Actual Behavior**:
The package does not load the config file under Linux.

**Comment by ArturDorochowicz** [comment]
For comparison, the config files in the NetAnalyzers package (https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers/6.0.0) are all lower case (see https://nuget.info/packages/Microsoft.CodeAnalysis.NetAnalyzers/6.0.0). That's probably how it should be fixed here.

**Comment by Youssef1313** [comment]
This was supposed to be fixed in https://github.com/dotnet/roslyn/pull/56021.

**Comment by Youssef1313** [comment]
The fix should be here I think:

https://github.com/dotnet/roslyn/blob/259a9e27ab8c6a1d8d3a3445b079fd5e3a13ec0a/src/CodeStyle/Tools/Program.cs#L73

**Comment by ArturDorochowicz** [comment]
As far as I understand, after dotnet/sdk#22918 (fixes dotnet/sdk#22870) the same files are part of the SDK as well as the package. It's possible that the upcoming SDK (6.0.200?) is affected by this issue as well, but I don't know if this is available publicly and have not checked it.

**Comment by Youssef1313** [comment]
I'm feeling that dotnet/sdk#22918 is unrelated to this issue, but @jmarolf can confirm for sure.

**Comment by ArturDorochowicz** [comment]
Description in dotnet/sdk#22918 explicitly lists mixed case editorconfig files as included due to it: 

> Here is a view of what the directory structure in the SDK looks like with this change
│   ├───build
│   │   │   Microsoft.CodeAnalysis.CSharp.CodeStyle.targets
│   │   │
│   │   └───config
│   │           AnalysisLevelStyle_All.editorconfig
│   │           AnalysisLevelStyle_Default.editorconfig
│   │           AnalysisLevelStyle_Minimum.editorconfig
│   │           AnalysisLevelStyle_None.editorconfig
│   │           AnalysisLevelStyle_Recommended.editorconfig

So unless the fix from #58957 is included in 6.0.200, it seems to me that the upcoming 6.0.200 could in fact be affected by this issue.

**Comment by Youssef1313** [comment]
@ArturDorochowicz In the SDK side, nothing special needs to be done. This will flow into the SDK whenever they use roslyn version that has the fix. The version update process is automated via PRs like this https://github.com/dotnet/sdk/pull/23234. Whether or not this will make into 6.0.200 is something I can't answer.

@jmarolf Can you confirm if this will flow automatically into the SDK, and which version of the SDK will get the fix?

**Comment by jmarolf** [comment]
At this point the 6.0.200 SDK is very near release. I am working to get this approved for 6.0.200, but considering it is not a regression (it has been present in all previous SDK releases) it may not make it. In that case the fix will flow into 6.0.300.

**Comment by ArturDorochowicz** [comment]
Versions 6.0.100, 6.0.101 were broken with regard to style analyzers, because they didn't have the necessary editorconfig files at all. Now 6.0.200 will have the files, but the feature may still be broken due to mixed file casing. That would be disappointing. I'm keeping my fingers crossed it will make it into 6.0.200.

When you say that if it doesn't get included in 6.0.200, that the next chance is 6.0.300, do you mean that there's no chance for it in like 6.0.20**1**? In other words, that it will need to wait for next _feature_ version? When would that be?

**Comment by jmarolf** [comment]
6.0.201 is only for serious build breaks for data loss.
